x/axelarnet/keeper/msg_server.go

```diff
@@ -511,22 +507,26 @@ func (s msgServer) ExecuteMessage(c context.Context, req *types.ExecuteMessageRe
 		return nil, fmt.Errorf("general message %s already executed", req.ID)
 	}
 
-	bz, err := types.TranslateMessage(msg, req.Payload)
-	if err != nil {
-		return nil, sdkerrors.Wrap(err, "invalid payload")
-	}
+	// send ibc message if destination is cosmos
+	if msg.Recipient.Chain.Module == exported.ModuleName {
```

Comment:
```suggestion
	if msg.Recipient.Chain.IsFrom(exported.ModuleName) {
```
same thing, but more readable.